src/imap.rs

```diff
@@ -1137,12 +1069,18 @@ impl Imap {
             )
             .await?;
 
+        let session = self.session.as_mut().context("no session")?;
+
         for (folder, rowid_set, uid_set) in UidGrouper::from(rows) {
-            self.select_folder(context, Some(&folder))
+            session
+                .select_folder(context, Some(&folder))
                 .await
                 .context("failed to select folder")?;
 
-            if let Err(err) = self.add_flag_finalized_with_set(&uid_set, "\\Seen").await {
+            if let Err(err) = session
+                .add_flag_finalized_with_set(&uid_set, "\\Seen")
+                .await
+            {
                 warn!(
```

Comment:
Btw, AFAIU rustfmt defaults for such cases are
```
    warn!(
        context,
        "Cannot move message, fallback to COPY/DELETE {} to {}: {}", set, target, err,
    );
```
Don't we want to switch to those defaults? Personally i have just 57 LOCs on my screen and prefer less extra line breaks

Reply:
Experimented locally with rustfmt. If all parameters don't fit on two (!) lines, it really wants them to be on separate lines. Why two, i don't know, some alien logic